src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -139,6 +139,15 @@ public OneOrMany<TResult> Select<TResult>(Func<T, TResult> selector)
                 OneOrMany.Create(_many.SelectAsArray(selector, arg));
         }
 
+        public IEnumerable<TResult> OfType<TResult>()
+        {
```

Comment:
Given how specialized this type is for single elements did you consider specializing this as well? Seems like you could make the one case a lot more effecient by doing something like: 

```csharp
public IEnumerable<TResult> OfType<TResult>()
{
  if (Count == 0) 
    return Array.Empty<TResult>();
  if (Count == 1) 
  {
    return this[0] is TResult result
      ? new TResult[] { result }
      : Array.Empty<TResult>();
  }

  return ManyCase();
  IEnumerable<TResult> ManyCase() 
  {
     foreach (var item in this)
     {
       if (item is TResult result) 
         yield return result;
     }
  }
}
```

Reply:
So this helper is *only* used by unit tests :D.  So i didn't care.

Reply:
Souds good